noir-projects/aztec-nr/aztec/src/macros/storage.nr

```diff
@@ -135,62 +146,3 @@ pub comptime fn storage_no_init(s: TypeDefinition) {
         f"Only one of #[storage] and #[storage_no_init] can be applied to the Storage struct.",
     );
 }
-
-/// Returns the expression required to initialize a state variable with a given slot, along with its serialization size,
-/// i.e. how many contiguous storage slots the variable requires.
-comptime fn generate_storage_field_constructor(typ: Type, slot: Quoted) -> (Quoted, u32) {
-    assert(
-        typ.as_data_type().is_some(),
-        "Storage containers must be generic structs of the form `Container<_, Context>`, or Map<Key, _, Context>",
-    );
-    let (container_struct, generics) = typ.as_data_type().unwrap();
-    let struct_name = container_struct.name();
-
-    let constructor = if is_storage_map(typ) {
-        // Map state variables recursively initialize their contents - this includes nested maps.
-        let (value_constructor, _) =
-            generate_storage_field_constructor(generics[1], quote { slot });
-
-        quote { $struct_name::new(context, $slot, | context, slot | { $value_constructor }) }
-    } else {
-        // We assume below that all state variables implement `fn new<Context>(context: Context, slot: Field) -> Self`.
-        quote { $struct_name::new(context, $slot)}
-    };
-
-    (constructor, get_storage_size(typ))
-}
-
-/// Returns true if `typ` is `state_vars::map::Map`.
-comptime fn is_storage_map(typ: Type) -> bool {
-    if typ.as_data_type().is_some() {
-        let (def, generics) = typ.as_data_type().unwrap();
-        let maybe_map = if (def.name() == quote { Map }) & (generics.len() == 3) {
-            let maybe_key = generics[0];
-            let maybe_value = generics[1];
-            let maybe_context = generics[2];
-            quote { crate::state_vars::map::Map<$maybe_key, $maybe_value, $maybe_context> }.as_type()
-        } else {
-            quote {()}.as_type()
-        };
-        typ == maybe_map
-    } else {
-        false
-    }
-}
-
-comptime fn add_context_generic(typ: Type, context_generic: Type) -> Type {
-    let (def, mut generics) = typ.as_data_type().expect(
-        f"Storage containers must be generic structs of the form `Container<..., Context>`",
-    );
-    let name = def.name();
-
-    if is_storage_map(typ) {
-        generics[generics.len() - 2] = add_context_generic(generics[1], context_generic);
-        generics[generics.len() - 1] = context_generic;
-    } else {
-        generics[generics.len() - 1] = context_generic;
-    }
-
-    let generics = generics.map(|typ: Type| quote {$typ}).join(quote {,});
-    quote { $name<$generics> }.as_type()
-}
```

Comment:
This is currently unused but would be relevant for implementing https://github.com/AztecProtocol/aztec-packages/issues/8658. Put there a [comment](https://github.com/AztecProtocol/aztec-packages/issues/8658#issuecomment-3608951047) such that this doesn't get lost if it's ever needed again.